src/common/vitals/vital-metric.js

```diff
@@ -0,0 +1,55 @@
+export class VitalMetric {
+  #values = []
+  #subscribers = new Set()
+  entries = []
+
+  constructor (name, roundingMethod) {
+    this.name = name
+    this.attrs = {}
+    this.roundingMethod = typeof roundingMethod === 'function' ? roundingMethod : Math.floor
+  }
+
+  update ({ value, entries = [], attrs = {}, addConnectionAttributes = false }) {
+    this.#values.push(this.roundingMethod(value))
+    this.entries.push(...entries)
```

Comment:
Might've mentioned this already but you may want to keep the entries segregated like `.push(entries)` so value is kept 1:1 with each **set**. Spreading entries make them indistinguishable.

Reply:
In the future, we may have callbacks provided to `subscribe` that cares about a set of PerformanceEntry's for a/each particular value derived from such set. I can think of such use case from some questions posed by Cheryl or UI so far in recent history.

Reply:
Let's address that in the future if needed and not treat it as a blocker since that would require basically a full rewrite of the PR for something that doesn't exist in the agent today

Reply:
But all that means is just `this.entries.push(entries)` and you've separated them?

Reply:
Let's defer this as entries are not really used or sent to NR atm, but the history of values still need addressing.